### PR TITLE
Add Cookidoo integration

### PR DESCRIPTION
**Pull request by miaucl**
## Proposed change

Add documentation for the new cookidoo integration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/129800
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/6037
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced Cookidoo integration for Home Assistant, enabling management of shopping lists.
	- Users can authenticate with their Cookidoo accounts and perform actions on their shopping lists.

- **Documentation**
	- Enhanced integration documentation with detailed instructions on managing Cookidoo shopping lists, including essential metadata and security recommendations.
	- Added related documentation links for further guidance.
	- Updated metadata to include new categories, IoT class, release version, configuration flow support, and code owners.

### CODE REVIEW

File: source/_integrations/cookidoo.markdown [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    title: Cookidoo the official Thermomix recipe platform
+  - url: https://www.vorwerk.com/
+    title: Vorwerk GmbH
+---
+
+The **Cookidoo** integration allows you to interact with your shopping lists of [Cookidoo the official Thermomix recipe platform](https://cookidoo.international/) within Home Assistant.
+
+For authentication, the integration requires the `email` and `password` you used for your Cookidoo account.
+
+{% details "Security recommendation" %}
```
Comment (zweckj) [inline source/_integrations/cookidoo.markdown]: i'd rather use {% important % }

File: source/_integrations/cookidoo.markdown [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+title: Cookidoo
+description: Instructions on how to integrate the Cookidoo todo list with Home Assistant.
+ha_category:
+  - To-do list
+ha_iot_class: Cloud Polling
+ha_release: 2024.12
```
[the remark this answers is not in the document]
Comment (miaucl) [reply source/_integrations/cookidoo.markdown]: ?

**Comment by miaucl** [comment]
@zweckj I have copied what I saw in `tedee` regarding the removal of an integration, but the builder complains about it.

**Comment by zweckj** [comment]
> @zweckj I have copied what I saw in `tedee` regarding the removal of an integration, but the builder complains about it.

Rebase 😉

File: source/_integrations/cookidoo.markdown [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+For example, if you add a pasta recipe, ingredients like "500g pasta" and "2 tomatoes" will appear in your `Shopping list`. You can check these items off as you shop, but you cannot modify the label.
+
+In contrast, in your `Additional purchases` list, you can freely add items like "Kitchen towels" or "Dish soap", and modify or remove them as needed.
+
+## Known Limitations
+
+{% note %}
+The Home Assistant to-do list interface allows both renaming items and changing their state. However, for the `Shopping list`, only state changes (checking/unchecking items) are supported. Any attempts to rename items will not be saved.
+{% endnote %}
```
Comment (klaasnicolaas) [inline source/_integrations/cookidoo.markdown]: Don't think it's necessary to put this in a note

File: source/_integrations/cookidoo.markdown [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    title: List of to-do list integrations
+  - docs: /dashboards/todo-list/
+    title: To-do list card
+  - url: https://cookidoo.international/
+    title: Cookidoo the official Thermomix recipe platform
+  - url: https://www.vorwerk.com/
+    title: Vorwerk GmbH
+---
+
+The **Cookidoo** integration allows you to interact with your shopping lists of [Cookidoo the official Thermomix recipe platform](https://cookidoo.international/) within Home Assistant.
```
Comment (klaasnicolaas) [inline source/_integrations/cookidoo.markdown]:
```suggestion
The **Cookidoo** {% term integration %} allows you to interact with your shopping lists of [Cookidoo the official Thermomix recipe platform](https://cookidoo.international/) within Home Assistant.
```

File: source/_integrations/cookidoo.markdown [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+For authentication, the integration requires the `email` and `password` you used for your Cookidoo account.
```
Comment (klaasnicolaas) [inline source/_integrations/cookidoo.markdown]: Use `{% configuration_basic %}` since I assume users enter an email and password in the config flow.

**Review by klaasnicolaas — DISMISSED** [review]

File: source/_integrations/cookidoo.markdown (hunk 3 above)
Comment (miaucl) [reply source/_integrations/cookidoo.markdown]: @klaasnicolaas Was a suggestion from coderabbit :)

**Comment by miaucl** [comment]
> > @zweckj I have copied what I saw in `tedee` regarding the removal of an integration, but the builder complains about it.
> 
> Rebase 😉

@zweckj I am up to date, rebased 5 hours ago.

**Comment by zweckj** [comment]
> > > @zweckj I have copied what I saw in `tedee` regarding the removal of an integration, but the builder complains about it.
> > 
> > 
> > Rebase 😉
> 
> @zweckj I am up to date, rebased 5 hours ago.

Ah, then it might just be on `current`, but not on `next`. Remove and add it later, or in a follow up PR.

**Comment by miaucl** [comment]
> > > > @zweckj I have copied what I saw in `tedee` regarding the removal of an integration, but the builder complains about it.
> > > 
> > > 
> > > Rebase 😉
> > 
> > 
> > @zweckj I am up to date, rebased 5 hours ago.
> 
> Ah, then it might just be on `current`, but not on `next`. Remove and add it later, or in a follow up PR.

@zweckj Strange, as the tedee integration has it also on next.

--> Rebuild of devcontainer did the job :)

File: source/_integrations/cookidoo.markdown [hunk 6]
```diff
@@ -0,0 +1,77 @@
+---
+title: Cookidoo
+description: Instructions on how to integrate the Cookidoo todo list with Home Assistant.
+ha_category:
+  - To-do list
+ha_iot_class: Cloud Polling
+ha_release: 2024.12
```
[the remark this answers is not in the document]
Comment (miaucl) [reply source/_integrations/cookidoo.markdown]: nope

File: source/_integrations/cookidoo.markdown [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+ha_config_flow: true
+ha_codeowners:
+  - '@miaucl'
+ha_domain: cookidoo
+ha_integration_type: service
+ha_platforms:
+  - todo
```
[the remark this answers is not in the document]
Comment (miaucl) [reply source/_integrations/cookidoo.markdown:14]: not yet

File: source/_integrations/cookidoo.markdown [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    title: Vorwerk GmbH
+---
+
+The **Cookidoo** {% term integration %} allows you to interact with your shopping lists of [Cookidoo the official Thermomix recipe platform](https://cookidoo.international/) within Home Assistant.
+
+{% configuration_basic %}
+Email:
+    description: "Enter the email address associated with your Cookidoo account to connect it to Home Assistant."
+Password:
+    description: "Enter the password for your Cookidoo account to enable the connection with Home Assistant."
```
Comment (frenck) [inline source/_integrations/cookidoo.markdown]: Simplified it a little, as connecting to Home Assistant is already kinda implied considering this is the Home Assistant documentation 😉 

```suggestion
Email:
    description: "Enter the email address associated with your Cookidoo."
Password:
    description: "Enter the password for your Cookidoo account."
```

File: source/_integrations/cookidoo.markdown [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+   - Contains ingredients from recipes
+   - Items can only be `checked`
+   - Items cannot be `created`, `deleted`, or `renamed`
+
+2. **Additional purchases**
+   - Contains user-added items to purchase
+   - Items can be `created`, `deleted`, and `updated`
+   - Items do not have a `description` field
+
+For example, if you add a pasta recipe, ingredients like `500g pasta` and `2 tomatoes` will appear in your `Shopping list`. You can check these items off as you shop, but you cannot modify the label.
```
Comment (frenck) [inline source/_integrations/cookidoo.markdown]:
```suggestion
For example, if you add a pasta recipe, ingredients like "500g pasta" and "2 tomatoes" will appear in your "Shopping list". You can check these items off as you shop, but you cannot modify the label.
```

File: source/_integrations/cookidoo.markdown [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+In contrast, in your `Additional purchases` list, you can freely add items like `Kitchen towels` or `Dish soap`, and modify or remove them as needed.
```
Comment (frenck) [inline source/_integrations/cookidoo.markdown]:
```suggestion
In contrast, in your "Additional purchases" list, you can freely add items like "Kitchen towels" or "Dish soap", and modify or remove them as needed.
```

File: source/_integrations/cookidoo.markdown [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+## Known Limitations
+
+{% important %}
+As Cookidoo cannot share shopping lists between accounts and everybody interacting with it uses the same credentials, make sure you protect your credentials accordingly. All users of your Home Assistant instance will have access to the same Cookidoo account.
+{% endimportant %}
+
+The Home Assistant to-do list interface allows both renaming items and changing their state. However, for the `Shopping list`, only state changes (checking/unchecking items) are supported. Any attempts to rename items will not be saved.
```
Comment (frenck) [inline source/_integrations/cookidoo.markdown]:
```suggestion
The Home Assistant to-do list interface allows both renaming items and changing their state. However, for the "Shopping list", only state changes (checking/unchecking items) are supported. Any attempts to rename items will not be saved.
```

File: source/_integrations/cookidoo.markdown [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+## Defining a custom polling interval
+
+The default polling interval is set to `90s`. This interval balances timely updates with server load. You might want to adjust this based on how frequently your shopping lists are updated.
+
+{% include common-tasks/define_custom_polling.md %}
```
Comment (frenck) [inline source/_integrations/cookidoo.markdown]: Let's format this as we have set in our integration quality scale documentation: https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/docs-data-update

**Review by frenck — DISMISSED** [review]

File: source/_integrations/cookidoo.markdown (hunk 12 above)
Comment (miaucl) [reply source/_integrations/cookidoo.markdown]: @frenck Therefore the `common-tasks/define_custom_polling.md` should not be used anymore?

Comment (miaucl) [reply source/_integrations/cookidoo.markdown]: I simplified it like in the post.

**Review by silamon — APPROVED** [review]

**Pull request merged**